### PR TITLE
[GENX] Verify DPAS input matrices dimensions

### PR DESCRIPTION
**Pull request by whitneywhtsang**

### CODE REVIEW

File: mlir/lib/Dialect/LLVMIR/IR/GENXOps.cpp [hunk 1]
```diff
@@ -23,20 +23,39 @@ using namespace mlir;
...
+  Type AElemTy = ATy.getElementType();
+  Type BElemTy = BTy.getElementType();
+  Type CElemTy = CTy.getElementType();
+  if (AElemTy != BElemTy)
+    return this->emitOpError(
+        "element type of 2nd (A) and 3rd (B) operands must match");
+
+  if (ATy.getNumElements() * AElemTy.getIntOrFloatBitWidth() != getRc() * 16)
+    return this->emitOpError(
+        "2nd operand (A) size should be repeat count times 16");
```
Comment (pengtu) [inline mlir/lib/Dialect/LLVMIR/IR/GENXOps.cpp]: 2nd operand (A) bit-size

File: mlir/lib/Dialect/LLVMIR/IR/GENXOps.cpp [hunk 2]
```diff
@@ -23,20 +23,39 @@ using namespace mlir;
...
+
+  if (BTy.getNumElements() * BElemTy.getIntOrFloatBitWidth() != 8 * 32)
+    return this->emitOpError(
+        "3rd operand (B) size should be systolic depth (8) times 16");
```
Comment (pengtu) [inline mlir/lib/Dialect/LLVMIR/IR/GENXOps.cpp]: 3rd operand (B) bit-size should be systolic depth (8) times 32

Comment (whitneywhtsang) [reply mlir/lib/Dialect/LLVMIR/IR/GENXOps.cpp]: opps

File: mlir/lib/Dialect/LLVMIR/IR/GENXOps.cpp [hunk 3]
```diff
@@ -23,20 +23,39 @@ using namespace mlir;
...
+  Type BElemTy = BTy.getElementType();
+  Type CElemTy = CTy.getElementType();
+  if (AElemTy != BElemTy)
+    return this->emitOpError(
+        "element type of 2nd (A) and 3rd (B) operands must match");
+
+  if (ATy.getNumElements() * AElemTy.getIntOrFloatBitWidth() != getRc() * 16)
+    return this->emitOpError(
+        "2nd operand (A) bit-size should be repeat count times 16");
+
```
Comment (etiotto) [inline mlir/lib/Dialect/LLVMIR/IR/GENXOps.cpp:56]: [nit]: constexpr unsigned SD = 8u;
Then use SD ...

Comment (etiotto) [reply mlir/lib/Dialect/LLVMIR/IR/GENXOps.cpp:56]: Add a comment here to explain the current requirements imposed by the IGC implementation

**Review by etiotto — APPROVED** [review]

**Review by pengtu — APPROVED** [review]

**Comment by pengtu** [comment]
Nice work!

**Pull request merged**